orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/SavePipelineTask.java

```diff
@@ -99,4 +106,18 @@ public long getBackoffPeriod() {
   public long getTimeout() {
     return TimeUnit.SECONDS.toMillis(30);
   }
+
+  private Map<String, Object> fetchExistingPipeline(Map<String, Object> newPipeline) {
+    String applicationName = (String) newPipeline.get("application");
+    String newPipelineID = (String) newPipeline.get("id");
+    if (newPipelineID != null && !newPipelineID.equals("")) {
+      List<Map<String, Object>> applicationPipelines = front50Service.getPipelines(applicationName);
+      for (Map<String, Object> existingPipeline : applicationPipelines) {
```

Comment:
Could use a java 8 `stream()` with `findFirst()` over a manual loop here.

Reply:
The stream api is a little daunting at first:

```java
.stream()
.filter(p -> p.containsKey("id"))
.filter(p -> p.get("id").equals(newPipelineID))
.findFirst().orElse(null)
```